content/en/blog/_posts/2023-08-04-kubernetes-1.28-blog.md

```diff
@@ -0,0 +1,275 @@
+fully independent of other indexes.
+
+For instance, if indexed jobs were used as the basis for a suite of long-running integration tests, then each test run would only be able to find a single test failure.
+
+    
+For more information, read [Handling Pod and container failures](/docs/concepts/workloads/controllers/job/#handling-pod-and-container-failures) in the Kubernetes documentation.
+   
+## CRI container and pod statistics without cAdvisor
+
+This encompasses two related pieces of work (changes to the kubelet's `/metrics/cadvisor` endpoint, and improvements to the replacement _summary_ API).
+
+There are two main APIs that consumers use to gather stats about running containers and pods: summary API and `/metrics/cadvisor`. The Kubelet is responsible for implementing the summary API, and cadvisor is responsible for fulfilling `/metrics/cadvisor`.
+
+This enhances CRI implementations to be able to fulfill all the stats needs of Kubernetes. At a high level, there are two pieces of this:
+
+- It enhances the CRI API with enough metrics to supplement the pod and container fields in the summary API directly from CRI.
+    
+- It enhances the CRI implementations to broadcast the required metrics to fulfill the pod and container fields in the `/metrics/cadvisor` endpoint.
+
+## Feature graduations and deprecations in Kubernetes v1.28
+### Graduations to stable
+
+This release includes a total of 12 enhancements promoted to Stable:
+
+* [`kubectl events`](https://github.com/kubernetes/enhancements/issues/1440)
+* [Retroactive default StorageClass assignment](https://github.com/kubernetes/enhancements/issues/3333)
+* [Non-graceful node shutdown](https://github.com/kubernetes/enhancements/issues/2268)
+* [Support 3rd party device monitoring plugins](https://github.com/kubernetes/enhancements/issues/606)
+* [Auth API to get self-user attributes](https://github.com/kubernetes/enhancements/issues/3325)
+* [Proxy Terminating Endpoints](https://github.com/kubernetes/enhancements/issues/1669)
+* [Expanded DNS Configuration](https://github.com/kubernetes/enhancements/issues/2595)
+* [Cleaning up IPTables Chain Ownership](https://github.com/kubernetes/enhancements/issues/3178)
+* [Minimizing iptables-restore input size](https://github.com/kubernetes/enhancements/issues/3453)
+* [Graduate the kubelet pod resources endpoint to GA](https://github.com/kubernetes/enhancements/issues/3743)
+* [Extend podresources API to report allocatable resources](https://github.com/kubernetes/enhancements/issues/2403)
+* [Move EndpointSlice Reconciler into Staging](https://github.com/kubernetes/enhancements/issues/3685)
+
+### Deprecations And Removals
+
+Removals:
+
+* [Removal of CSI Migration for GCE PD](https://github.com/kubernetes/enhancements/issues/1488)
+
+Deprecations:
+
+* [Ceph RBD in-tree plugin](https://github.com/kubernetes/kubernetes/pull/118303)
+* [Ceph FS in-tree plugin](https://github.com/kubernetes/enhancements/pull/4035)
```

Comment:
https://github.com/kubernetes/kubernetes/pull/118143 is right link
```suggestion
* [Ceph FS in-tree plugin](https://github.com/kubernetes/kubernetes/pull/118143)
```

Reply:
We could expand “FS”, too.